### ISSUE TITLE
New curly configuration proposal

### ISSUE THREAD

**Issue by inikulin** [eslint/eslint#2890]
Currently curly rule has two configurations:
   *  *multi* - bodies with single statement shouldn't be wrapped;
   *  *multi-line* - one-liners shouldn't be wrapped.

I would like to suggest another rule: 
> If body contains single statement it shouldn't be wrapped. But if this single statement is multiline then it should be wrapped.

The following patterns are considered warnings:
```js
if (foo)
    return {
        bar: baz,
        qux: foo
    };

while (true)
  if(foo)
      doSomething();
  else
      doSomethingElse();

if (foo) {
    return;
}

while (true) {
    doSomething();
}

for (var i=0; i < items.length; i++) {
    doSomething();
}
```

The following patterns are not considered warnings:
```js
if (foo) {
    return {
        bar: baz,
        qux: foo
    };
}

while (true) {
  if(foo)
      doSomething();
  else
      doSomethingElse();
}

if (foo) 
    return;

while (true) 
    doSomething();

for (var i=0; i < items.length; i++) 
    doSomething();
```

I can issue a PR for this functionality if it will be accepted.

**Comment by gyandeeps** [comment]
Possibly related to https://github.com/eslint/eslint/issues/1806

**Comment by inikulin** [comment]
Yep, seems like a dup. But I don't see if original issue was accepted. If so I can issue a PR.

**Comment by nzakas** [comment]
Please continue discussion on #1806
